revocation/crl/cache/cache.go

```diff
@@ -0,0 +1,56 @@
+// Copyright The Notary Project Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cache provides methods for caching CRL
+//
+// The fileSystemCache is an implementation of the Cache interface that uses the
+// file system to store CRLs. The file system cache is built on top of the OS
+// file system to leverage the file system's concurrency control and atomicity.
+//
+// The CRL is stored in a tarball format, which contains two files: base.crl and
+// metadata.json. The base.crl file contains the base CRL in DER format, and the
+// metadata.json file contains the metadata of the CRL.
+//
+// To implement a new cache, you need to create a new struct that implements the
+// Cache interface.
+//
+// > Note: Please ensure that the implementation supports *CRL as the
+// type of value field to cache a CRL.
+package cache
+
+import (
+	"context"
+	"time"
+)
+
+const (
+	// DefaultMaxAge is the default maximum age of the CRLs cache.
+	// If the CRL is older than DefaultMaxAge, it will be considered as expired.
+	//
+	// reference: Baseline Requirements for Code-Signing Certificates
+	// 4.9.7 CRL issuance frequency: https://cabforum.org/uploads/Baseline-Requirements-for-the-Issuance-and-Management-of-Code-Signing.v3.9.pdf
+	DefaultMaxAge = 7 * 24 * time.Hour
```

Comment:
Shouldn't this be governed by expiry in CRL? If yes, why do we need default?

Reply:
According to 
> [5.1.2.5](https://www.rfc-editor.org/rfc/rfc5280#section-5.1.2.5).  Next Update
   This field indicates the date by which the next CRL will be issued.
   The next CRL could be issued before the indicated date, but it will
   not be issued any later than the indicated date.

The new CRL may be published before the next update. Also, the next update is an optional field in the CRL. We need a default one to force the update.